pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,311 @@
+package cgroup
+
+import (
+	"bufio"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/aquasecurity/tracee/pkg/mount"
+)
+
+//
+// Constants
+//
+
+const (
+	procCgroups     = "/proc/cgroups"
+	procFilesystems = "/proc/filesystems"
+
+	CgroupV1FsType = "cgroup"
+	CgroupV2FsType = "cgroup2"
+
+	DefaultCgroupController = "cpuset"
+
+	TmpCgroup       = "/tmp"
+	TmpCgroupV1Path = "tracee-cgroupv1"
+	TmpCgroupV2Path = "tracee-cgroupv2"
+)
+
+type CgroupVersion int
+
+const (
+	CgroupV1 CgroupVersion = iota + 1
+	CgroupV2
+)
+
+//
+// Cgroups
+//
+
+// Cgroups handles cgroups initialization (v1+v2 or v2 only)
+type Cgroups struct {
+	cgroupv1 *Cgroup // used for cgroup filesystem readings
+	cgroupv2 *Cgroup // used for cgroup prog attachment (and readings, if v2 is default)
+	cgroup   *Cgroup // default cgroup (might point to one or another)
+}
```

Comment:
Why do we need to keep both cgroupv1 and cgroupv2? Won't having the deafult cgroup be enough?

Reply:
I think this one takes care of the case where you have a hybrid system with both.

Reply:
I believe this is solved (Nadav explained the reason).